src/sync/all.rs

```diff
@@ -372,14 +372,25 @@ impl<TRq, TSrc, TBl> AllSync<TRq, TSrc, TBl> {
                 let outer_source_id_entry = self.shared.sources.vacant_entry();
                 let outer_source_id = SourceId(outer_source_id_entry.key());
 
-                let source_id = all_forks.add_source(
-                    AllForksSourceExtra {
-                        user_data,
-                        outer_source_id,
-                    },
-                    best_block_number,
-                    best_block_hash,
-                );
+                let source_user_data = AllForksSourceExtra {
+                    user_data,
+                    outer_source_id,
+                };
+
+                let source_id =
+                    match all_forks.prepare_add_source(best_block_number, best_block_hash) {
+                        all_forks::AddSource::BestBlockAlreadyVerified(b) => {
```

Comment:
can we combine these 3 branches?

Reply:
Can't combine all three, as the types are different